lib/xgu/nv2a_regs.h

```diff
@@ -0,0 +1,1321 @@
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_DST_ALPHA           0x0304
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_ONE_MINUS_DST_ALPHA 0x0305
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_DST_COLOR           0x0306
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_ONE_MINUS_DST_COLOR 0x0307
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_SRC_ALPHA_SATURATE  0x0308
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_CONSTANT_COLOR      0x8001
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_ONE_MINUS_CONSTANT_COLOR 0x8002
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_CONSTANT_ALPHA      0x8003
+#       define NV097_SET_BLEND_FUNC_DFACTOR_V_ONE_MINUS_CONSTANT_ALPHA 0x8004
+#   define NV097_SET_BLEND_COLOR                              0x0000034C
+#   define NV097_SET_BLEND_EQUATION                           0x00000350
+#       define NV097_SET_BLEND_EQUATION_V_FUNC_SUBTRACT           0x800A
+#       define NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT   0x800B
+#       define NV097_SET_BLEND_EQUATION_V_FUNC_ADD                0x8006
+#       define NV097_SET_BLEND_EQUATION_V_MIN                     0x8007
+#       define NV097_SET_BLEND_EQUATION_V_MAX                     0x8008
+#       define NV097_SET_BLEND_EQUATION_V_FUNC_REVERSE_SUBTRACT_SIGNED 0xF005
+#       define NV097_SET_BLEND_EQUATION_V_FUNC_ADD_SIGNED         0xF006
+#   define NV097_SET_DEPTH_FUNC                               0x00000354
+#   define NV097_SET_COLOR_MASK                               0x00000358
+#       define NV097_SET_COLOR_MASK_BLUE_WRITE_ENABLE             (1 << 0)
+#       define NV097_SET_COLOR_MASK_GREEN_WRITE_ENABLE            (1 << 8)
+#       define NV097_SET_COLOR_MASK_RED_WRITE_ENABLE              (1 << 16)
+#       define NV097_SET_COLOR_MASK_ALPHA_WRITE_ENABLE            (1 << 24)
+#   define NV097_SET_DEPTH_MASK                               0x0000035C
+#   define NV097_SET_STENCIL_MASK                             0x00000360
+#   define NV097_SET_STENCIL_FUNC                             0x00000364
+#   define NV097_SET_STENCIL_FUNC_REF                         0x00000368
+#   define NV097_SET_STENCIL_FUNC_MASK                        0x0000036C
+#   define NV097_SET_STENCIL_OP_FAIL                          0x00000370
+#   define NV097_SET_STENCIL_OP_ZFAIL                         0x00000374
+#   define NV097_SET_STENCIL_OP_ZPASS                         0x00000378
+#       define NV097_SET_STENCIL_OP_V_KEEP                        0x1E00
+#       define NV097_SET_STENCIL_OP_V_ZERO                        0x0000
+#       define NV097_SET_STENCIL_OP_V_REPLACE                     0x1E01
+#       define NV097_SET_STENCIL_OP_V_INCRSAT                     0x1E02
+#       define NV097_SET_STENCIL_OP_V_DECRSAT                     0x1E03
+#       define NV097_SET_STENCIL_OP_V_INVERT                      0x150A
+#       define NV097_SET_STENCIL_OP_V_INCR                        0x8507
+#       define NV097_SET_STENCIL_OP_V_DECR                        0x8508
```

Comment:
Create an enum for all of these (like `XguStencilOp` with values like `XGU_KEEP`, `XGU_ZERO` etc.), and use the new enum type for the parameter for the `NV097_SET_STENCIL_OP_*` functions.

Reply:
Added. :+1: